### PR TITLE
Build with Docker documentation

### PR DESCRIPTION
**Pull request by staiyeba**
Start of build with docker documentation.

### CODE REVIEW

File: Building-with-Docker.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+When building the docker image there are a few options available:
+
++-----------------------+--------------------------------------------+
+|       **Action**      |                  **Command**               |
+|-----------------------|--------------------------------------------|
+| Specify build version | --build-arg TAG=<git tag/version to build> |
+|-----------------------|--------------------------------------------|
+|   Service to build    | --target=<build/grubify/webserver>         |
+|-----------------------|--------------------------------------------|
+|      Custom Tag       |  --build-arg TAG=<custom-tag>              |
```
Comment (mnordsletten) [inline Building-with-Docker.rst]: This one has been renamed to `CUSTOM_TAG`. And you should put in what it does: 
> Makes any image built with this docker container self-identify with this custom tag.

File: Building-with-Docker.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
++-----------------------+--------------------------------------------+
+
+Docker tag structure
+~~~~~~~~~~~~~~~~~~~~
+
+The docker tags in use for these images are:
+
+::
+
+<org>/<service>:<IncludeOS_Tag>.<Dockerfile version> includeos/build:v0.12.0-rc.3.1
```
Comment (mnordsletten) [inline Building-with-Docker.rst]: You can remove the `dockerfile version` part of this.

**Review by mnordsletten — COMMENTED** [review]
We don't need to show examples of using the `CUSTOM_TAG` option. As this is not that important, and can stay a "hidden" feature. We will mention it in the list of build options, but that is it. 

Other than that we should better document the different build targets. base, source-build, grubify, build. What does the various stages accomplish?

Then the main focus of the documentation should be on using the container in build mode. How to use it to build your service.

Also we can add documentation on the `/ios_version.txt` file. Which is kinda important.

**Comment by staiyeba** [comment]
Thank you. I will review and get back.

**Comment by staiyeba** [comment]
This PR needs to be updated again after we have finished with our conan ci setup. Therefore closing this PR for now.

**Pull request closed without merging**